apps/src/lib/node/ledger/events/log/dumb_queries.rs

```diff
@@ -0,0 +1,155 @@
+//! Silly simple Tendermint query parser.
+//!
+//! This parser will only work with simple queries of the form:
+//!
+//! ```text
+//! tm.event='NewBlock' AND <accepted|applied>.<$attr>='<$value>'
+//! ```
+
+use lazy_static::lazy_static;
+use regex::Regex;
+
+use crate::node::ledger::events::{Event, EventType};
+
+/// Regular expression used to parse Tendermint queries.
+const QUERY_PARSING_REGEX_STR: &str =
+    r"^tm\.event='NewBlock' AND (accepted|applied)\.([\w_]+)='([^']+)'$";
+
+lazy_static! {
+    /// Compiled regular expression used to parse Tendermint queries.
+    static ref QUERY_PARSING_REGEX: Regex = Regex::new(QUERY_PARSING_REGEX_STR).unwrap();
+}
+
+/// A [`QueryMatcher`] verifies if a Namada event matches a
+/// given Tendermint query.
+#[derive(Debug, Clone)]
+pub struct QueryMatcher<'q> {
+    event_type: EventType,
+    attr: String,
+    value: &'q str,
+}
+
+impl<'q> QueryMatcher<'q> {
+    /// Checks if this [`QueryMatcher`] validates the
+    /// given [`Event`].
+    pub fn matches(&self, event: &Event) -> bool {
+        event.event_type == self.event_type
+            && event
+                .attributes
+                .get(&self.attr)
+                .map(|value| value == self.value)
+                .unwrap_or_default()
+    }
+
+    /// Parses a Tendermint-like events query.
+    pub fn parse(query: &'q str) -> Option<Self> {
+        let captures = QUERY_PARSING_REGEX.captures(query)?;
+
+        let event_type = match captures.get(1)?.as_str() {
+            "accepted" => EventType::Accepted,
+            "applied" => EventType::Applied,
+            // NOTE: the regex only matches `accepted`
+            // and `applied`
+            _ => unreachable!(),
+        };
+        let attr = captures.get(2)?.as_str().to_string();
+        let value = captures.get(3)?.as_str();
+
+        Some(Self {
+            event_type,
+            attr,
+            value,
+        })
+    }
+}
```

Comment:
Re. https://github.com/anoma/namada/pull/607#discussion_r996798254 do we need the regex matching anymore? We can directly construct the matchers now (could even just be called `EventMatcher` or smth)

Reply:
I can remove this in the next PR https://github.com/anoma/namada/pull/610